test/routing/ArrivalsRouterSpec.scala

```diff
@@ -48,49 +48,56 @@ class ArrivalsRouterSpec extends AnyFreeSpec with Matchers with OptionValues wit
     new FakeV2ArrivalsController()
   )
 
-  "when creating a Arrival Notification" - {
-    "with accept header set to application/vnd.hmrc.2.0+json (version two)" - {
+  for (
+    endpoint <- Seq(
+      routes.ArrivalsRouter.createArrivalNotification(),
+      routes.ArrivalsRouter.getArrival("123"),
+      routes.ArrivalsRouter.getArrivalsForEori()
+    )
+  )
+    s"${endpoint.method} ${endpoint.url}" - {
+      "with accept header set to application/vnd.hmrc.2.0+json (version two)" - {
 
-      val arrivalsHeaders = FakeHeaders(
-        Seq(HeaderNames.ACCEPT -> "application/vnd.hmrc.2.0+json", HeaderNames.CONTENT_TYPE -> "application/xml")
-      )
+        val arrivalsHeaders = FakeHeaders(
+          Seq(HeaderNames.ACCEPT -> "application/vnd.hmrc.2.0+json", HeaderNames.CONTENT_TYPE -> "application/xml")
+        )
 
-      "must route to the v2 controller and return Accepted when successful" in {
+        "must route to the v2 controller and return Accepted when successful" in {
 
-        val request =
-          FakeRequest(method = "POST", uri = routes.ArrivalsRouter.createArrivalNotification().url, body = <test></test>, headers = arrivalsHeaders)
-        val result = call(sut.createArrivalNotification(), request)
+          val request =
+            FakeRequest(method = "POST", uri = endpoint.url, body = <test></test>, headers = arrivalsHeaders)
+          val result = call(sut.createArrivalNotification(), request)
```

Comment:
I see you're trying to check that all methods lead to the same place (as per the `for` loop above, but you're then only calling this one method on the actual router, and this is the key thing that needs to change.

Reply:
Good spot, thanks Dan